packages/fetchai/skills/weather_station/strategy.py

```diff
@@ -115,9 +121,10 @@ def generate_proposal_and_data(
                 "seller_tx_fee": self._seller_tx_fee,
                 "currency_id": self._currency_id,
                 "ledger_id": self._ledger_id,
+                "tx_nonce": tx_nonce if tx_nonce is not None else "",
```

Comment:
Ledger_apis can return None in case it fails for some reason. There is a try / except block in the implementation.

Reply:
Ok but if it fails, we should not send a proposal, so the proposal should be None and we need to adjust the logic to aboard the negotiation.

Reply:
See above outstanding comment, the proposal should only be created if `tx_nonce` is not None. But in general, I cannot see why it would ever be None. The return type of `generate_tx_nonce` is not Optional! So just have `"tx_nonce": tx_nonce`